packages/react/src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -83,40 +81,24 @@ export const ModalBody = React.forwardRef<HTMLDivElement, ModalBodyProps>(
   ) {
     const prefix = usePrefix();
     const contentRef = useRef<HTMLDivElement>(null);
-    const [isScrollable, setIsScrollable] = useState(false);
+
+    const { height } = useResizeObserver({ ref: contentRef });
+
+    const isScrollable =
+      !!contentRef.current &&
+      contentRef?.current?.scrollHeight > contentRef?.current?.clientHeight;
```

Comment:
* `isScrollable` implicitly relies on `height`. We could make that dependency more obvious for the future with a useMemo.
* With the double negate in place we should be able to remove the optional chaining.

```suggestion
    const { height } = useResizeObserver({ ref: contentRef });

    const isScrollable = React.useMemo(() => {
      const node = contentRef.current;
      return !!node && node.scrollHeight - node.clientHeight;
    }, [height, children]); // Would including children here make sense?
```

Reply:
@tay1orjones 
I tested this locally and `eslint` throws an error, it considers the dependencies unnecessary

`493:6  warning  React Hook useMemo has unnecessary dependencies: 'children' and 'height'. Either exclude them or remove the dependency array  react-hooks/exhaustive-deps`

Should I just add a comment block explaining the dependency instead?